src/mesh/unstructured_mesh.C

```diff
@@ -993,6 +993,41 @@ void UnstructuredMesh::find_neighbors (const bool reset_remote_elements,
       }
   }
 
+#ifdef LIBMESH_ENABLE_PERIODIC
+  auto * db = this->get_disconnected_boundaries();
+
+  if (db)
+    {
+      // Obtain a point locator
+      std::unique_ptr<PointLocatorBase> point_locator = this->sub_point_locator();
+
+      // Get the disconnected boundaries object (from periodic BCs)
+      auto * db = this->get_disconnected_boundaries();
+
+      for (const auto & element : this->element_ptr_range())
+        {
+          for (auto ms : element->side_index_range())
+            {
+              // Skip if this side already has a valid neighbor (including remote neighbors)
+              if (element->neighbor_ptr(ms) != nullptr &&
+                  element->neighbor_ptr(ms) != remote_elem)
+                continue;
+
+                for (const auto & [id, boundary_ptr] : *db)
+                {
+                  unsigned int neigh_side;
+                  const Elem * neigh = db->neighbor(id, *point_locator, element, ms, &neigh_side, true /*skip_found_check*/);
```

Comment:
What's the use case for `skip_found_check` here?  In general we prefer error messages over wrong answers, and it looks like here we expect to see a neighbor, and with `skip_found_check == false` any violation of that expectation would give us an error message, and with `skip_found_check == true` (plus any kernel depending on that neighbor pointer) we'd expect a wrong answer.

I'd figure the only hard case here would be the distributed-mesh "neighbor is remote" case, but that's supposed to be handled by the case where it's already `remote_elem` because we got those links right already (either a priori or in `delete_remote_elements`).